### PR TITLE
[OTEL-2125] Add E2E test for no dd exporter case

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?
Add an E2E test case for OTel agent with no datadog exporter config.

### CODE REVIEW

File: test/new-e2e/tests/otel/otel-agent/no_dd_exporter_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+type noDDExporterTestSuite struct {
+	e2e.BaseSuite[environments.Kubernetes]
+}
+
+//go:embed config/no-dd-exporter.yml
+var noDDExporterConfig string
+
+func (s *noDDExporterTestSuite) SetupSuite() {
+	s.BaseSuite.SetupSuite()
+}
```
Comment (songy23) [inline test/new-e2e/tests/otel/otel-agent/no_dd_exporter_test.go:30]:
```suggestion
```

File: cmd/otel-agent/subcommands/run/command.go [hunk 2]
```diff
@@ -106,8 +107,20 @@ func runOTelAgentCommand(ctx context.Context, params *subcommands.GlobalParams,
...
+			// TODO: don't rely on this pattern; remove this `OptionalModuleWithParams` thing
+			//       and instead adapt OptionalModule to allow parameter passing naturally.
+			//       See: https://github.com/DataDog/datadog-agent/pull/28386
+			configsyncimpl.OptionalModuleWithParams(),
+			fx.Provide(func() configsyncimpl.Params {
+				return configsyncimpl.NewParams(params.SyncTimeout, params.SyncDelay, true)
+			}),
 			converterfx.Module(),
-			fx.Provide(func(cp converter.Component) confmap.Converter {
+			fx.Provide(func(cp converter.Component, _ optional.Option[configsync.Component]) confmap.Converter {
```
Comment (songy23) [inline cmd/otel-agent/subcommands/run/command.go:123]: There are needed to set up `configsync`

Comment (liustanley) [reply cmd/otel-agent/subcommands/run/command.go:123]: Could you add more context for these changes and the change in the ddflareextension in the PR description, or move these to a new PR?

Comment (songy23) [reply cmd/otel-agent/subcommands/run/command.go:123]: makes sense, filed https://github.com/DataDog/datadog-agent/pull/32363

Comment (songy23) [reply cmd/otel-agent/subcommands/run/command.go]: merged & rebased

**Review by liustanley — APPROVED** [review]

**Pull request merged**